tests/installation/ipxe_install.pm

```diff
@@ -89,12 +89,12 @@ sub set_bootscript {
     }
 
     my $cmdline_extra;
-    $cmdline_extra .= " regurl=$regurl " if $regurl;
+    $cmdline_extra .= " regurl=$regurl " if ($regurl and !is_usb_boot);
```

Comment:
I do not think you should make this change, because you never really install with these parameters typed in in ```ipxe_install.pm```.

Reply:
These two lines:

$cmdline_extra .= " regurl=$regurl " if ($regurl and !is_usb_boot);
$cmdline_extra .= " root=/dev/ram0 initrd=initrd textmode=1" if (check_var('IPXE_UEFI', '1') and !is_usb_boot);

affect SelfInstall ISO host installation ? How ? I think you just boot SelfInstall ISO directly without modify any boot parameters.

Reply:
Please read below comment which is also given in later code. ipxe_install.pm is used to bring up *15sp5 GM* ipxe install, to serve usb installation later for sle micro 6 selfinstall. So regurl which is passed down from medium as sle micro proxy scc shall not be used here for 15sp5 gm, but will be used in latter test steps after sle micro installation for system registration. Also `" root=/dev/ram0 initrd=initrd textmode=1"` is not needed for usb boot.

```
    # SLE Micro 6.0 Self-Install image does not directly support pxe boot.
    # To install it on bare metal machine, firstly bring up a minimum system via ipxe
    # with this function(eg sle15sp5 gm). But we do not need to finish installation,
    # booting to sshd-server-started is enough, at which we will have a ssh console
    # to do latter steps.
    # Then dd the Self-Install iso to a USB device.
    # And then boot from the USB, and finish installation with the Self-Install iso.
    # For more details, refer to poo#151498.
```

Reply:
The description does not answer my concern.

Step1. Set IPXE boot and boot SUT from IPXE with 15-SP5 medium. All params are used for this medium only.
Step2. When ssh server is up. SSH to SUT and configure usb boot.
Step3. Reboot SUT and boot from SLE Micro 6.0 medium in USB device. This does not use $cmdline_extra I think.

So how $cmdline_extra affects SLE Micro 6.0 medium installlation ???

Reply:
> The description does not answer my concern.
> 
> Step1. Set IPXE boot and boot SUT from IPXE with 15-SP5 medium. All params are used for this medium only. Step2. When ssh server is up. SSH to SUT and configure usb boot. Step3. Reboot SUT and boot from SLE Micro 6.0 medium in USB device. This does not use $cmdline_extra I think.
> 
> So how $cmdline_extra affects SLE Micro 6.0 medium installlation ???

$cmdline_extra does not affect  SLE Micro 6.0 medium installlation, but it affect step 1 15sp5 pxe boot. Both 15sp5 and SLE Micro 6.0 are *now* having the same settings passed from openqa-cli for final purpose of sle micro self install iso installation.

Reply:
15-SP5 iIPXE boot does not change compared with prior to this pull request (15-SP5 iPXE installation works well on OSD) and settings here does not affect fresh SLE Micro 6.0 installation either. So I do not think you should make these changes here.

Reply:
> 15-SP5 iIPXE boot does not change compared with prior to this pull request (15-SP5 iPXE installation works well on OSD) and settings here does not affect fresh SLE Micro 6.0 installation either. 

It seems to me you still did not get the point. Let me try to rephrase ...

This has nothing to do with the current or future regular 15sp5 installation on OSD. Neither does SLE Micro 6.0 installation on qemu backend. But the 15sp5 gm ipxe boot that is *used in USB_BOOT test* shall be different from the regular 15sp5 installation on OSD.

In the usb installation method, openqa triggers a job for sle micro selfinstall iso. So the settings are sle micro scc url, sle micro iso, USB_INSTALL etc, but no sle15sp5 GM settings at all. But within *this job*, we do firstly 15sp5 ipxe boot(until sshd starts) *via the ipxe_boot.pm*. If not filtering for USB_BOOT, this  ipxe_boot.pm would generate $cmdline_extra and put it into the ipxe boot menu of the test machine on ipxe's http server. This is NOT needed. So  such conditions are added to differentiate the NORMAL SLES IPXE installation from the 15sp5 GM ipxe installation used by usb boot.

Reply:
But I do not think scc_url affects usb boot, because you do not need it at all, you just need ssh server up. 15-SP5 installation never starts and never tries to contact scc url passed in.